batch-inference/batch-inference-transcript-summarization.ipynb

```diff
@@ -0,0 +1,516 @@
+    "\n",
+    "This notebook demonstrates how to leverage batch inference for summarizing call center transcripts at scale. By processing substantial volumes of text transcripts in batches. Though we are using example of call transcript summarization here, you can really apply this to any other use case that does not need a real time output.\n",
+    "\n",
+    "## Prerequisites\n",
+    "\n",
+    "Before you begin, ensure that you have the following prerequisites in place:\n",
+    "1. Updated boto3 to 1.35.1 or greater version\n",
+    "2. Have raw data stored in S3 bucket as \n",
+    "3. Permissions to invoke `create_model_invocation_job` API. Refer to the documentation to learn about [required permissions for batch inference job](https://docs.aws.amazon.com/bedrock/latest/userguide/batch-inference-permissions.html).\n",
+    "4. Permission to read and write data on Amazon S3 bucket.\n",
+    "5. Call transcript dataset:\n",
+    "* This notebook was built using synthetic call transcripts in `.txt` files. If you want to try it with your own dataset, upload your call transcripts to an Amazon S3 bucket in `.txt` format. Each text file in the S3 bucket should contain only one call transcript.\n",
+    "* If you do not have a dataset but want to try out Batch Inference for Amazon Bedrock, you can use the synthetic call data available [here](https://github.com/aws-samples/amazon-bedrock-samples/batch-inference/dataset/synthetic_call_transcript_data.zip). You need to unzip the data, and then upload the `.txt` files to an S3 bucket to use the notebook below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0f646a1c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# upgrade boto3 \n",
+    "%pip install --upgrade pip\n",
+    "%pip install boto3 --upgrade"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dff5840d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# restart kernel\n",
+    "from IPython.core.display import HTML\n",
+    "HTML(\"<script>Jupyter.notebook.kernel.restart()</script>\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1321d9c0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import boto3\n",
+    "import json\n",
+    "import os\n",
+    "from datetime import datetime\n",
+    "import time\n",
+    "\n",
+    "# Bedrock client for batch inference job\n",
+    "bedrock = boto3.client(service_name=\"bedrock\")\n",
+    "\n",
+    "# Create an S3 client\n",
+    "s3 = boto3.client('s3')\n",
+    "\n",
+    "# # Set the S3 bucket name and prefix for the text files\n",
+    "bucket_name = '<your-s3-bucket-name>'\n",
+    "raw_data_prefix = '<raw-data-prefix>' + '/'\n",
+    "out_prefix = '<prefix-for-batch-output>'\n",
+    "\n",
+    "# Batch API parameters:\n",
+    "role_Arn = \"<ARN of the role with permissions to invoke batch api for amazon bedrock>\"\n",
```

Comment:
This variable seems not being used by the following code, can delete?

Reply:
ok, in the following code this variable was called "roleArn"

Reply:
maybe we can use the following code, as we've already asked in the pre-requisite to have create_model_invocation_job permission?:

from sagemaker import get_execution_role
roleArn = get_execution_role()